### ISSUE TITLE
Proper action chain testing

### ISSUE THREAD

**Issue by satispunk** [mobxjs/mobx#441]
Hi,

I am trying to test store actions using sinon

```javascript

class Store {
}
@action action2() {
}
@action action1() {
    ...
    this.action2()
}
```

Is there a way to understand if action2 is called whenever action1 is called?

Tried to use sinon
```
sinon.stub(store, 'action2', () => {})
```
but got TypeError: Cannot redefine property: action2

**Comment by satispunk** [comment]
found a workaround

```javascript
action1 = action(() => {})
```

**Comment by mweststrate** [comment]
Note that `action1 = action(...)`  is not the same as `@action action1() {}`, the first declares an instance property, the latter a prototype property. I've just made a custom build `mobx@2.4.0-fix441`, would you mind checking whether that fixes the original problem?

**Comment by satispunk** [comment]
@mweststrate 
I will try it,

I guess there is no difference for me between proptotype and instance

**Comment by satispunk** [comment]
@mweststrate 

Seems it works for both

@action someName() {}
@action someName = () => {}

**Comment by mweststrate** [comment]
Released the fix as part of 2.4.1
